dcmgr/lib/dcmgr/endpoints/12.03/instances.rb

```diff
@@ -186,8 +186,15 @@ def set_monitoring_parameters(instance)
 
   get '/:instance_id/password' do
     @instance = find_by_uuid(:Instance, params[:instance_id])
-    respond_with(R::InstancePassword.new(@instance).generate)
-  end
+    theresponse = R::InstancePassword.new(@instance).generate
+
+    if Dcmgr::Configurations.dcmgr.windows.delete_password_on_request
+      logger.info "Deleting encrypted_password for #{:instance_id}"
+      @instance.set({ :encrypted_password => nil }).save_changes
```

Comment:
it does not have to remove the password in GET. it is just enough to return nil value in the response hash data.

since you have two different version of set password nil, here misses to set nil to ``:password_will_be_deleted_at``.
please make the method in Models::Instance to clear them.